src/components/MembershipTypeButton.tsx

```diff
@@ -0,0 +1,49 @@
+"use client";
+
+import React from "react";
+
+import { cn } from "@/lib/utils";
+
+interface MembershipOptionProps {
+  selectedMembership: boolean | undefined;
+  onChange: (isMember: boolean) => void;
+  typeHeading: string;
+  description1: string;
+  description2: string;
+}
+
+export const MembershipTypeButton = ({
+  selectedMembership,
+  onChange,
+  typeHeading,
+  description1,
+  description2,
+}: MembershipOptionProps) => (
+  <label
+    className={cn(
+      "flex",
+      "flex-col",
+      "py-[21px]",
+      "px-8",
+      "py-6",
+      "shadow",
+      "rounded",
+      "hover:pointer",
+      "cursor-pointer",
+      selectedMembership
+        ? "bg-primary text-secondary hover:bg-primary/90"
```

Comment:
text should be color `primary-foreground`
```suggestion
        ? "bg-primary text-primary-foreground hover:bg-primary/90"
```